extensions/libraries/redcore/model/list.php

```diff
@@ -378,7 +378,7 @@ protected function populateState($ordering = null, $direction = null)
 		// Keep B/C for components previous to jform forms for filters
 		{
 			// Pre-fill the limits
-			$defaultLimit = $params ? $params->get('list_limit', $app->get('list_limit')) : $app->get('list_limit');
+			$defaultLimit = ($params && $params->get('list_limit') >= 0) ? $params->get('list_limit', $app->get('list_limit')) : $app->get('list_limit');
```

Comment:
why is this restriction for >= 0? has someone been setting it to negative value?